### PR TITLE
69

### PR DESCRIPTION
**Pull request by mil-lesya**

### CODE REVIEW

File: server/tests/transaction.test.ts [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+const newTransaction = {
+  transactionType: TransactionType.EXPENSE,
+  category: 'New',
+  place: 'Mac',
+  price: 11,
+  currency: 'USD',
+  description: 'Good dinner'
+}
+
+let token: string;
```
Comment (ivanjermakov) [inline server/tests/transaction.test.ts:34]: move to `describe` callback body (line :37)

File: server/tests/transaction.test.ts [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    Transaction.deleteMany({}).then(done)
+    Category.deleteMany({}).then(done)
+  });
+
+  it('should use test database', () => {
+    expect(config.MONGO_URI).toBe(process.env.MONGO_URI_TEST);
+  });
+
+  it('should create new transaction', done => {
+    http
```
Comment (ivanjermakov) [inline server/tests/transaction.test.ts:66]: Verify created transaction?

File: server/tests/transaction.test.ts [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+      .post('/api/transaction/new')
```
Comment (ivanjermakov) [inline server/tests/transaction.test.ts]: use `transactionRoute` from `app.ts`

Comment (mil-lesya) [reply server/tests/transaction.test.ts]: no full route name in the `app.ts`

Comment (ivanjermakov) [reply server/tests/transaction.test.ts]: wym? `${transactionRoute}/new`

https://github.com/mil-lesya/expense-tracker/blob/1b761d933b8491080f6820367f15c2db5f8d9719/server/app.ts#L16

File: server/routes/api/transaction.ts [hunk 4]
```diff
@@ -12,42 +12,46 @@ const router = Router();
...
+        category,
+        place: transaction.place,
+        price: transaction.price,
+        currency: transaction.currency
       });
+
+      newTransaction.save().then((transaction: ITransaction) => {
+        if (!transaction) res.status(400);
+        socket.emit('new', newTransaction);
+        res.status(200);
```
Comment (ivanjermakov) [inline server/routes/api/transaction.ts]: Replace this with `res.status(200).json()` to force response return.
As far as I can tell, without explicit response return, tests somehow skips this line or something.

File: server/tests/transaction.test.ts [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+  currency: 'USD',
+  description: 'Good dinner'
+}
+
+let token: string;
+
+describe('transactions test', () => {
+
+  beforeAll(done => {
+    request(app)
```
Comment (ivanjermakov) [inline server/tests/transaction.test.ts]: Replace with `http`

**Pull request merged**